common/src/main/java/com/wynntils/mc/render/FontRenderer.java

```diff
@@ -213,6 +230,15 @@ private float calculateRenderHeight(List<TextRenderTask> toRender) {
         return (float) (height - NEWLINE_OFFSET / 2 * McUtils.window().getGuiScale());
```

Comment:
This looks sus. I think this is the reason the vertically centered text gets incorrectly offset. Why would you subtract half the newline offset multiplied with the gui scale? I think some parentheses are missing.

Reply:
Also, the for loop (github won't let me comment on those lines) need some failsafe, so it does not just stack up `lines` and reset i to parts.size() if it fails to get a sublist. (I think that is what is causing the infinite looping.)

Reply:
https://github.com/Wynntils/Artemis/pull/172/files#diff-4b2bbac5a5d922ca9738a301beb33915427e526a4cf0e22f97ec98cebbc46c5d for the second comment.